tests/functional/setups/manu_thermoporomech_nofrac_2d.py

```diff
@@ -0,0 +1,1219 @@
+
+    def fluid_source(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Fluid source."""
+
+        # Internal sources are inherit from parent class
+        # Zero internal sources for unfractured domains, but we keep it to maintain
+        # good code practice
+        internal_sources: pp.ad.Operator = super().fluid_source(subdomains)
+
+        # External sources are retrieved from pp.TIME_STEP_SOLUTIONS and wrapped as an
+        # AdArray.
+        external_sources = pp.ad.TimeDependentDenseArray(
+            name="source_flow",
+            domains=self.mdg.subdomains(),
+            previous_timestep=True,
+        )
+
+        # Add up contribution of internal and external sources of fluid
+        fluid_sources = internal_sources + external_sources
+        fluid_sources.set_name("Fluid source")
+
+        return fluid_sources  # - prod
+
+    def body_force(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Body force."""
+
+        external_sources = pp.ad.TimeDependentDenseArray(
+            name="source_mechanics",
+            domains=self.mdg.subdomains(),
+            previous_timestep=True,
+        )
+
+        return external_sources
+
+    def energy_source(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Thermal source."""
+
+        # Internal sources are inherit from parent class
+        # Zero internal sources for unfractured domains, but we keep it to maintain
+        # good code practice
+        internal_sources: pp.ad.Operator = super().energy_source(subdomains)
+
+        # External sources are retrieved from pp.TIME_STEP_SOLUTIONS and wrapped as an
+        # AdArray.
+        external_sources = pp.ad.TimeDependentDenseArray(
+            name="source_energy",
+            domains=self.mdg.subdomains(),
+            previous_timestep=True,
+        )
+
+        # Add up contribution of internal and external sources of fluid
+        thermal_sources = internal_sources + external_sources
+        thermal_sources.set_name("Energy source")
+
+        return thermal_sources
+
+
+# -----> Solution strategy
+class ManuThermoPoroMechSolutionStrategy2d(
+    pp.thermoporomechanics.SolutionStrategyThermoporomechanics
+):
+    """Solution strategy for the verification setup."""
+
+    exact_sol: ManuThermoPoroMechExactSolution2d
+    """Exact solution object."""
+
+    fluid: pp.FluidConstants
+    """Object containing the fluid constants."""
+
+    results: list[ManuThermoPoroMechSaveData]
+    """List of SaveData objects."""
+
+    def __init__(self, params: dict):
+        """Constructor for the class."""
+        super().__init__(params)
+
+        self.exact_sol: ManuThermoPoroMechExactSolution2d
+        """Exact solution object."""
+
+        self.results: list[ManuThermoPoroMechSaveData] = []
+        """Results object that stores exact and approximated solutions and errors."""
+
+        self.flux_variable: str = "darcy_flux"
```

Comment:
I don't see why this and the stress variable are not class variables. That's the common practice in the models.